tests/test_metadata.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+"""Test cases for "metadata.py". """
+
+import copy
+import unittest
+
+from securesystemslib.metadata import Envelope
+from securesystemslib.signer import Signature
+
+
+class TestEnvelope(unittest.TestCase):
+    """Test metadata interface provided by DSSE envelope."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.signature_dict = {
+            "keyid": "11fa391a0ed7a447cbfeb4b2667e286fc248f64d5e6d0eeed2e5e23f97f9f714",
+            "sig": "30460221009342e4566528fcecf6a7a5d53ebacdb1df151e242f55f8775883469cb01dbc6602210086b426cc826709acfa2c3f9214610cb0a832db94bbd266fd7c5939a48064a851",
+        }
+        cls.envelope_dict = {
+            "payload": "aGVsbG8gd29ybGQ=",
+            "payloadType": "http://example.com/HelloWorld",
+            "signatures": [cls.signature_dict],
+        }
+        cls.pae = b"DSSEv1 29 http://example.com/HelloWorld 11 hello world"
+
+    def test_envelope_from_to_dict(self):
+        """Test envelope to_dict and from_dict methods"""
+
+        # create envelope object from its dict.
+        envelope_obj = Envelope.from_dict(copy.deepcopy(self.envelope_dict))
```

Comment:
`envelope_dict` contains a nested dict and these tests changes the signature, so I have to use `deepcopy`.